### PR TITLE
Add option to hide plugin setting from auto-generated plugin setting page

### LINKED ISSUE #4591: [FR] Allow to hide plugin settings
### Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find a similar feature request

### Problem statement

I want to build my own select component on the settings page of my plugin and handle user input validation myself (Using the settings content mixin). This is already possible thought the `get_setting` and `set_setting` methods, but the input ist still shown on the automatically generated setting page for the plugin. (related to #4546 and wolflu05/inventree-cups-plugin#1)

### Suggested solution

Add an extra `HIDE` parameter to the settings definition in the plugin and check that before rendering here: https://github.com/inventree/InvenTree/blob/0b6e2ee592b885183ee6bf5f70b94a904b0caae3/InvenTree/templates/InvenTree/settings/mixins/settings.html#L12-L14 

### Describe alternatives you've considered

*ignore that the input is still shown*

### Examples of other systems

_No response_

### Do you want to develop this?

- [ ] I want to develop this.

### PR DESCRIPTION
**Pull request by wolflu05**
Please let me know if there is something else where we need to change something and if we should name the new `hide` setting more precise like `hideFromSettingPage`.

fixes #4591

### CODE REVIEW

**Comment by SchrodingersGat** [comment]
I like the simple approach but would prefer `hidden` (attribute / adjective) rather than `hide` (action / verb).

**Review by SchrodingersGat — APPROVED** [review]

**Comment by SchrodingersGat** [comment]
Nice. Can you find somewhere in the documentation to add a quick note about this feature? Otherwise it will be quite undiscoverable for plugin devs

File: docs/docs/extend/plugins/settings.md [hunk 1]
```diff
@@ -52,6 +52,11 @@ class PluginWithSettings(SettingsMixin, InvenTreePlugin):
                 MaxValueValidator(25)
             ]
         },
+        'HIDDEN_SETTING': {
+            'name': _('Hidden Setting'),
+            'description': _('This setting is hidden from the automatically generated plugin settings page'),
+            'hidden': True,
+        }
```
Comment (wolflu05) [inline docs/docs/extend/plugins/settings.md:59]: @SchrodingersGat That's why I added this. You want me to add a small note somewhere else?

Comment (SchrodingersGat) [reply docs/docs/extend/plugins/settings.md:59]: Yeah a small callout near that would be good

```markdown
!!! tip "Hidden Settings"
    Plugin settings can be hidden from the settings page by marking them as 'hidden'
```

Comment (wolflu05) [reply docs/docs/extend/plugins/settings.md:59]: Added it. Do you know suggestions code blocks? That way we could just click on apply directly on GitHub.

Comment (SchrodingersGat) [reply docs/docs/extend/plugins/settings.md:59]: didn't know that, I'll keep it in mind :)

**Comment by SchrodingersGat** [comment]
Thanks for the contribution :)

**Pull request merged**